apps/meteor/app/lib/server/functions/relinquishRoomOwnerships.ts

```diff
@@ -1,17 +1,21 @@
 import { Messages, Roles, Rooms, Subscriptions, ReadReceipts } from '@rocket.chat/models';
 
 import { FileUpload } from '../../../file-upload/server';
+import { notifyOnSubscriptionChanged } from '../lib/notifyListener';
 import type { SubscribedRoomsForUserWithDetails } from './getRoomsWithSingleOwner';
 
-const bulkRoomCleanUp = async (rids: string[]): Promise<unknown> => {
-	// no bulk deletion for files
-	await Promise.all(rids.map((rid) => FileUpload.removeFilesByRoomId(rid)));
-
-	return Promise.all([
-		Subscriptions.removeByRoomIds(rids),
+const bulkRoomCleanUp = async (rids: string[]): Promise<void> => {
+	await Promise.all([
+		Subscriptions.removeByRoomIds(rids, {
+			async onTrash(doc) {
+				void notifyOnSubscriptionChanged(doc, 'removed');
+			},
+		}),
 		Messages.removeByRoomIds(rids),
 		ReadReceipts.removeByRoomIds(rids),
 		Rooms.removeByIds(rids),
+		// no bulk deletion for files
+		...rids.map((rid) => FileUpload.removeFilesByRoomId(rid)),
```

Comment:
IMHO, this change is not related to the watcher changes and should be in its own PR